products/iambeta/api.yaml

```
@@ -94,11 +94,199 @@ objects:
+          `projects/<projectnumber>/locations/global/workloadIdentityPools/<pool-id>`.
         output: true
       - !ruby/object:Api::Type::Boolean
         name: 'disabled'
         description: |
           Whether the pool is disabled. You cannot use a disabled pool to exchange tokens, or use
           existing tokens to access resources. If the pool is re-enabled, existing tokens grant
           access again.
+  - !ruby/object:Api::Resource
+    name: 'WorkloadIdentityPoolProvider'
+    min_version: beta
+    base_url: projects/{{project}}/locations/global/workloadIdentityPools/{{workload_identity_pool_id}}/providers
+    self_link: projects/{{project}}/locations/global/workloadIdentityPools/{{workload_identity_pool_id}}/providers/{{workload_identity_pool_provider_id}}
+    create_url: projects/{{project}}/locations/global/workloadIdentityPools/{{workload_identity_pool_id}}/providers/?workloadIdentityPoolProviderId={{workload_identity_pool_provider_id}}
+    update_verb: :PATCH
+    update_mask: true
+    description: A configuration for an external identity provider.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Managing workload identity providers':
+          'https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#managing_workload_identity_providers'
+      api: 'https://cloud.google.com/iam/docs/reference/rest/v1beta/projects.locations.workloadIdentityPools.providers'
+    properties:
+      - !ruby/object:Api::Type::String
+         name: 'workloadIdentityPoolId'
+         description: |
+           The ID used for the pool, which is the final component of the pool resource name. This
+           value should be 4-32 characters, and may contain the characters [a-z0-9-]. The prefix
+           `gcp-` is reserved for use by Google, and may not be specified.
+         required: true
+         input: true
+         url_param_only: true
+      - !ruby/object:Api::Type::String
+         name: 'workloadIdentityPoolProviderId'
+         description: |
+           The ID for the provider, which becomes the final component of the resource name. This
+           value must be 4-32 characters, and may contain the characters [a-z0-9-]. The prefix
+           `gcp-` is reserved for use by Google, and may not be specified.
+         required: true
+         input: true
+         url_param_only: true
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: |
+          The state of the provider.
+          STATE_UNSPECIFIED: State unspecified.
+          ACTIVE: The provider is active, and may be used to validate authentication credentials.
+          DELETED: The provider is soft-deleted. Soft-deleted providers are permanently deleted
+          after approximately 30 days. You can restore a soft-deleted provider using
+          UndeleteWorkloadIdentityPoolProvider. You cannot reuse the ID of a soft-deleted provider
+          until it is permanently deleted.
```

```suggestion
          The state of the provider.
          * STATE_UNSPECIFIED: State unspecified.
          * ACTIVE: The provider is active, and may be used to validate authentication credentials.
          * DELETED: The provider is soft-deleted. Soft-deleted providers are permanently deleted
          after approximately 30 days. You can restore a soft-deleted provider using
          UndeleteWorkloadIdentityPoolProvider. You cannot reuse the ID of a soft-deleted provider
          until it is permanently deleted.
```

Check the[ generated markdown for this ](https://github.com/modular-magician/terraform-provider-google/blob/3305dc4d2d598059156eff60940ea5aee8ee93c2/website/docs/r/iam_workload_identity_pool_provider.html.markdown)